mlrun/model_monitoring/db/tsdb/tdengine/tdengine_connector.py

```diff
@@ -251,7 +274,50 @@ def read_metrics_data(
             ],
         ],
     ]:
-        raise NotImplementedError
+        if type == "metrics":
+            table = mm_schemas.TDEngineSuperTables.METRICS
+            name = mm_schemas.MetricData.METRIC_NAME
+            df_handler = self.df_to_metrics_values
+        elif type == "results":
+            table = mm_schemas.TDEngineSuperTables.APP_RESULTS
+            name = mm_schemas.ResultData.RESULT_NAME
+            df_handler = self.df_to_results_values
+        else:
+            raise mlrun.errors.MLRunInvalidArgumentError(
+                f"Invalid type {type}, must be either 'metrics' or 'results'."
+            )
+
+        list_of_metrics = [
+            f"({mm_schemas.WriterEvent.APPLICATION_NAME} = '{metric.app}' AND {name} = '{metric.name}')"
+            for metric in metrics
+        ]
+        with StringIO() as query:
+            query.write(f"endpoint_id='{endpoint_id}' ")
+            query.write("AND ")
+            query.write(" OR ".join(list_of_metrics))
+            filter_query = query.getvalue()
```

Comment:
Here too, using `StringIO` is overkill, but it also seems to be wrong, because `AND` takes precedence over `OR`.

Maybe something like
```suggestion
        metrics_condition = " OR ".join([
            f"({mm_schemas.WriterEvent.APPLICATION_NAME} = '{metric.app}' AND {name} = '{metric.name}')"
            for metric in metrics
        ])
        filter_query = f"endpoint_id='{endpoint_id}' AND ({metrics_condition})"
```

Reply:
good catch!